Symfony/CS/Fixer/Symfony/FunctionTypehintSpaceFixer.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Symfony;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ */
+class FunctionTypehintSpaceFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        for ($index = $tokens->count() - 1; $index >= 0; --$index) {
+            $token = $tokens[$index];
+
+            if (!$token->isGivenKind(T_FUNCTION)) {
+                continue;
+            }
+
+            $startParenthesisIndex = $tokens->getNextTokenOfKind($index, array('('));
+            $endParenthesisIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $startParenthesisIndex);
+
+            for ($iter = $endParenthesisIndex - 1; $iter > $startParenthesisIndex; --$iter) {
+                if (!$tokens[$iter]->isGivenKind(T_VARIABLE)) {
+                    continue;
+                }
+
+                // skip ... before $variable for variadic parameter
+                if (defined('T_ELLIPSIS')) {
+                    $prevNonWhitespaceIndex = $tokens->getPrevNonWhitespace($iter);
+                    if ($tokens[$prevNonWhitespaceIndex]->isGivenKind(T_ELLIPSIS)) {
+                        $iter = $prevNonWhitespaceIndex;
+                    }
+                }
+
+                // skip & before $variable for parameter passed by reference
+                $prevNonWhitespaceIndex = $tokens->getPrevNonWhitespace($iter);
+                if ($tokens[$prevNonWhitespaceIndex]->equals('&')) {
+                    $iter = $prevNonWhitespaceIndex;
+                }
+
+                if (!$tokens[$iter - 1]->equalsAny(array(array(T_WHITESPACE), array(T_COMMENT), array(T_DOC_COMMENT), '(', ','))) {
```

Comment:
Agreed :) the mem. usage stays around the same because PHP will clean up the array after the function returns for the non-static var and only at the end of the script for the static one.
The second metric of interest is time, for example:

```php
<?php
$start = microtime(true);

function getCount() {
// dynamic
//    $count = count([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]);

// vs. static
    static $b = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20];
    $count = count($b);

    return $count;
}

for ($i = 0; $i < 2500000; ++$i) {
    getCount();
}

$end = microtime(true);

echo "\n\n";
echo memory_get_peak_usage();
echo "\n";
echo memory_get_peak_usage(true);
echo "\n";
echo sprintf('%5.5f', $end - $start);
echo "\n";
```

Results:

dynamic
242840
262144
5.51659
possum@sfx1:~/work/dbal$ php test.php 

static:
244776
262144
3.90172
possum@sfx1:~/work/dbal$ php test.php 

I admit, the gain is very small (microsecs on millions of operations).